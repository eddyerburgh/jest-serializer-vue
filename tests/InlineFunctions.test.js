const helpers = require('./helpers.js');

import { mount } from '@vue/test-utils';
import InlineFunctions from './components/InlineFunctions.vue';

describe('InlineFunctions.vue', () => {
  test('Functions kept', () => {
    helpers.mockSettings({
      clearInlineFunctions: false
    });
    let wrapper = mount(InlineFunctions);

    expect(wrapper)
      .toMatchSnapshot();
  });

  test('Functions removed', () => {
    helpers.mockSettings({
      clearInlineFunctions: true
    });
    let wrapper = mount(InlineFunctions);

    expect(wrapper)
      .toMatchSnapshot();
  });
});
