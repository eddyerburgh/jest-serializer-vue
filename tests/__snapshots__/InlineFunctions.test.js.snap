// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineFunctions.vue Functions kept 1`] = `
<div>
  <h1 title="function () {
        return true;
      }">Classic</h1>
  <h2 title="function named() {
        return true;
      }">Multi-line</h2>
  <h3 title="function (x) {
        return !x;
      }">Arrow 1</h3>
  <h3 title="function (x) {
        return !x;
      }">Arrow 2</h3>
  <h3 title="function () {
        return true;
      }">Arrow 3</h3>
  <h3 title="function (x, y) {
        return x + y;
      }">Arrow 4</h3>
  <h4 title="function () { [native code] }">Reference</h4>
  <h5 title="2">Run</h5>
  <h6 title="function (a word used in programming) is typically followed by curly braces, ie {}">Red herring</h6>
  <div><pre>      function notInAnAttribute (a, b, c) {
        const d = a * c;
        const e = d - b;
        return e;
      }
    </pre>
  </div>
  <div title="this is what a it looks like function () {}">
    Attribute does not start with function.
  </div>
  <div title="function () {} is what it looks like">
    Attribute does not end with }.
  </div>
  <p title="function () { [native code] },function () { [native code] }">In array</p>
</div>
`;

exports[`InlineFunctions.vue Functions removed 1`] = `
<div>
  <h1 title="[function]">Classic</h1>
  <h2 title="[function]">Multi-line</h2>
  <h3 title="[function]">Arrow 1</h3>
  <h3 title="[function]">Arrow 2</h3>
  <h3 title="[function]">Arrow 3</h3>
  <h3 title="[function]">Arrow 4</h3>
  <h4 title="[function]">Reference</h4>
  <h5 title="2">Run</h5>
  <h6 title="function (a word used in programming) is typically followed by curly braces, ie {}">Red herring</h6>
  <div><pre>      function notInAnAttribute (a, b, c) {
        const d = a * c;
        const e = d - b;
        return e;
      }
    </pre>
  </div>
  <div title="this is what a it looks like function () {}">
    Attribute does not start with function.
  </div>
  <div title="function () {} is what it looks like">
    Attribute does not end with }.
  </div>
  <p title="[function]">In array</p>
</div>
`;
